pkg/build/controller/controller.go

```diff
@@ -228,3 +229,71 @@ func (bc *BuildPodController) CancelBuild(build *buildapi.Build, pod *kapi.Pod)
 func isBuildCancellable(build *buildapi.Build) bool {
 	return build.Status == buildapi.BuildStatusNew || build.Status == buildapi.BuildStatusPending || build.Status == buildapi.BuildStatusRunning
 }
+
+// BuildPodDeleteController watches pods running builds and updates the build if the pod is deleted
+type BuildPodDeleteController struct {
+	BuildStore   cache.Store
+	BuildUpdater buildclient.BuildUpdater
+}
+
+func (bc *BuildPodDeleteController) HandleBuildPodDeletion(pod *kapi.Pod) error {
+	glog.V(4).Infof("Handling deletion of build pod %s/%s", pod.Namespace, pod.Name)
+	var build *buildapi.Build
+	for _, obj := range bc.BuildStore.List() {
+		b := obj.(*buildapi.Build)
+		if buildutil.GetBuildPodName(b) == pod.Name {
```

Comment:
GetBuildPodName is deterministic to the name of the build - so why aren't we just asking for the build with that name?